src/Spice86.Core/Emulator/VM/Breakpoint/BreakpointConditionParser.cs

```diff
@@ -0,0 +1,316 @@
+namespace Spice86.Core.Emulator.VM.Breakpoint;
+
+using Spice86.Core.Emulator.CPU.CfgCpu.Ast;
+using Spice86.Core.Emulator.CPU.CfgCpu.Ast.Operations;
+using Spice86.Core.Emulator.CPU.CfgCpu.Ast.Value;
+using Spice86.Core.Emulator.CPU.CfgCpu.Ast.Value.Constant;
+using System.Globalization;
+
+/// <summary>
+/// Parses breakpoint condition strings into CfgCpu AST nodes.
+/// Supports C-like expression syntax with operators, registers, and memory access.
+/// </summary>
+public class BreakpointConditionParser {
```

Comment:
Rename that to AstExpressionParser and move it to src/Spice86.Core/Emulator/CPU/CfgCpu/Ast/Parser/